apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -221,13 +221,14 @@ function enqueue_coblocks_gallery_scripts() {
  * Load Blog Posts block.
  */
 function load_blog_posts_block() {
-	$slug          = 'newspack-blocks/newspack-blocks.php';
+	// Use regex instead of static slug in order to match plugin installation also from github, where slug may contain (HASH|branch-name).
+	$slug_regex    = '/newspack-blocks(-[A-Za-z0-9-]+)?\/newspack-blocks\.php/';
 	$disable_block = (
 		( defined( 'WP_CLI' ) && WP_CLI ) ||
 		/* phpcs:ignore WordPress.Security.NonceVerification */
-		( isset( $_GET['action'], $_GET['plugin'] ) && 'activate' === $_GET['action'] && $slug === $_GET['plugin'] ) ||
-		in_array( $slug, (array) get_option( 'active_plugins', array() ), true ) ||
-		in_array( $slug, (array) get_site_option( 'active_sitewide_plugins', array() ), true )
+		( isset( $_GET['action'], $_GET['plugin'] ) && 'activate' === $_GET['action'] && preg_match( $slug_regex, esc_url_raw( wp_unslash( $_GET['plugin'] ) ) ) ) ||
```

Comment:
`$_GET['plugin']` is not a URL but rather a plugin slug. `esc_url_raw()` is an escaping function and not meant for sanitization. [`sanitize_text_field()`](https://developer.wordpress.org/reference/functions/sanitize_text_field/) might be more appropriate here